pkg/fs/local_file_system.go

```diff
@@ -0,0 +1,285 @@
+// Licensed to Apache Software Foundation (ASF) under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Apache Software Foundation (ASF) licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+// Package fs (file system) is an independent component to operate file and directory.
+package fs
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/apache/skywalking-banyandb/pkg/logger"
+)
+
+// LocalFileSystem implements the File System interface.
+type LocalFileSystem struct {
+	logger *logger.Logger
+}
+
+// LocalFile implements the File interface.
+type LocalFile struct {
+	file *os.File
+}
+
+// NewLocalFileSystem is used to create the Local File system.
+func NewLocalFileSystem() FileSystem {
+	return &LocalFileSystem{
+		logger: logger.GetLogger(moduleName),
+	}
+}
+
+func readErrorHandle(err error, name string, size int) (int, error) {
```

Comment:
I believe this method can handle all errors, but it would be to add a parameter indicating the operation where the error occurred.